srcpkgs/fatsort/template

```diff
@@ -0,0 +1,18 @@
+# Template file for 'fatsort'
+pkgname=fatsort
+version=1.6.2.605
+revision=1
+build_style=gnu-makefile
+hostmakedepends="help2man"
+short_desc="Utility that sorts FAT12, FAT16, FAT32 and exFAT partitions"
+maintainer="tibequadorian <tibequadorian@posteo.de>"
+license="GPL-2.0-or-later"
+homepage="https://fatsort.sourceforge.io/"
+distfiles="https://iweb.dl.sourceforge.net/project/fatsort/fatsort-${version}.tar.xz"
+checksum=08cafa36f86ab89e7a3241d437df9af8a008549d7d13416256b7aaa006d5ffb7
+nocross=yes
```

Comment:
This is only `nocross` because upstream requires you to generate a man page on the fly using `help2man` and the target executable can't be run on the host. You have two easy fixes to make this cross-compilable:
1. Place a pregenerated `fatsort.1` in the `files` subdirectory alongside the template, then add
```
cp ${FILESDIR}/fatsort.1 man 
```
in either `post_patch` (see comment below) or `pre_build`.

2. Add `build_helper="qemu"` to the template and add the following snippet to `post_patch`:
```
if [ "$CROSS_BUILD}" ]; then
	cat > src/fatsort.qemu <<-EOF
	#!/bin/sh
	qemu-${XBPS_TARGET_QEMU_MACHINE}-static ${wrksrc}/src/fatsort "\$@"
	EOF
	chmod 755 src/fatsort.qemu
	vsed -i man/Makefile -e 's|../src/fatsort|../src/fatsort.qemu|'
fi
```
This will dynamically generate the man page as upstream intends but will use qemu to make it cross-compilable.

Whatever you do, you should also file an issue upstream to get them to ship a pregenerated `fatsort.1` since there is no reason we should be dynamically generating it with every build.

Reply:
done, thanks for help!